### PR TITLE
Fix ClearML project paths

### LINKED ISSUE #24820: ClearML callback uses local output path `project` as ClearML `project_name`, causing failures with absolute paths
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

_No response_

### Bug

Ultralytics currently passes `trainer.args.project` directly to ClearML as `project_name`:

```python
task = Task.init(
    project_name=trainer.args.project or "Ultralytics",
    task_name=trainer.args.name,
    ...
)
```

This can cause problems because in Ultralytics, `project` is commonly used as a local filesystem output directory, for example:

```python
model.train(
    project="/path/to/output/runs/experiment_group",
    name="trial_0",
)
```

However, ClearML interprets `project_name` as a ClearML project name/path, not as a filesystem output path.

When `project` is an absolute path starting with `/`, the value is passed directly to ClearML as the ClearML project name. In my environment, this triggered a ClearML project lookup/create mismatch.

The lookup used the project name with a leading slash:

```text
name=/tmp/test/runs/optuna_202606012
```

but the existing project reported by ClearML had the leading slash removed:

```text
name=tmp/test/runs/optuna_202606012
```

This caused later task creation to fail with:

```text
Action failed <400/801: projects.create/v1.0 (Project with the same name already exists: name=tmp/test/runs/optuna_202606012, company=<COMPANY_ID>)> (name=/tmp/test/runs/optuna_202606012, description=, method=get)
```

### Minimal ClearML reproduction

This reproduces the underlying ClearML behavior:

```python
from clearml import Task


def main():
    for i in range(2):
        task = Task.init(
            project_name="/tmp/test/runs/optuna_202606012",
            task_name=f"repro_{i}",
            reuse_last_task_id=False,
        )
        task.close()


if __name__ == "__main__":
    main()
```

Removing the leading slash avoids the issue:

```python
project_name="tmp/test/runs/optuna_202606012"
```

### Why this affects Ultralytics

Ultralytics maps the training arguments to ClearML like this:

```python
ClearML project_name = trainer.args.project
ClearML task_name = trainer.args.name
```

So when users set `project` to an absolute output directory, the absolute filesystem path is sent to ClearML as the ClearML project name.

### Expected behavior

Ultralytics should avoid passing a local filesystem path directly as ClearML `project_name`, or it should sanitize the value before passing it to ClearML.

Possible minimal fix:

```python
project_name=str(trainer.args.project or "Ultralytics").lstrip("/")
```

This would separate the local training output directory from the ClearML project namespace.

### Affected versions

Confirmed in:

- `ultralytics==8.4.0`
- `ultralytics==8.4.66`

The current `8.4.66` ClearML callback still contains:

```python
project_name=trainer.args.project or "Ultralytics"
```

### Environment

- Ultralytics: `8.4.0`
- Also checked latest PyPI release: `8.4.66`
- ClearML Server: self-hosted Docker
- ClearML Server image: `clearml/server:latest`
- ClearML package inside server image: `2.1.3`
- Python: `3.10`

---

Ultralytics 8.4.0 🚀 Python-3.10.19 torch-2.10.0.dev20251208+cu128 CUDA:0 (NVIDIA GeForce RTX 3070, 7841MiB)
Setup complete ✅ (24 CPUs, 62.7 GB RAM, 47.1/231.2 GB disk)

OS                     Linux-5.15.0-174-generic-x86_64-with-glibc2.35
Environment            Linux
Python                 3.10.19
Install                pip
Path                    /conda/envs/yolo/lib/python3.10/site-packages/ultralytics
RAM                    62.67 GB
Disk                   47.1/231.2 GB
CPU                    AMD Ryzen 7 8700G w/ Radeon 780M Graphics
CPU count              24
GPU                    NVIDIA GeForce RTX 3070, 7841MiB
GPU count              2
CUDA                   12.8

numpy                  ✅ 2.2.6>=1.23.0
matplotlib             ✅ 3.10.8>=3.3.0
opencv-python          ✅ 4.12.0.88>=4.6.0
pillow                 ✅ 12.0.0>=7.1.2
pyyaml                 ✅ 6.0.3>=5.3.1
requests               ✅ 2.32.5>=2.23.0
scipy                  ✅ 1.15.3>=1.4.1
torch                  ✅ 2.10.0.dev20251208+cu128>=1.8.0
torch                  ✅ 2.10.0.dev20251208+cu128!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision            ✅ 0.25.0.dev20251208+cu128>=0.9.0
psutil                 ✅ 7.1.3>=5.8.0
polars                 ✅ 1.36.0>=0.20.0
ultralytics-thop       ✅ 2.0.18>=2.0.18

### Minimal Reproducible Example

```



from ultralytics import YOLO


def main():
    for i in range(2):
        model = YOLO("yolo11n.pt")
        model.train(
            data="coco8.yaml",
            epochs=1,
            imgsz=640,
            batch=8,
            project="/tmp/test/runs/optuna_202606012",
            name=f"repro_{i}",
            exist_ok=True,
            save=False,
            plots=False,
            val=False,
        )


if __name__ == "__main__":
    main()
````

### Additional

_No response_

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by MUKILAN0608**
I have read the CLA Document and I sign the CLA

## Summary

Fixes #24820.

Ultralytics currently passes `trainer.args.project` directly to ClearML as `project_name`.

When users provide a POSIX absolute output directory such as:

```python
project="/tmp/test/runs/experiment"
```

the value is forwarded directly to ClearML, which interprets it as a project namespace rather than a filesystem path. This can lead to project lookup/create inconsistencies.

## Changes

* Added `_clearml_project_name()` helper.
* Normalize leading POSIX-style slashes before passing values to ClearML.
* Preserve relative project names unchanged.
* Preserve Windows and UNC-style paths unchanged.
* Added regression tests covering:

  * None
  * empty string
  * relative paths
  * POSIX absolute paths
  * root path (`/`)
  * Windows paths
  * callback integration behavior

## Validation

* 14 regression tests passed.
* Ruff checks passed.
* Python compilation checks passed.

## CLA

I have read the CLA Document and I sign the CLA.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Fixes ClearML `project_name` handling for POSIX-style output paths so ClearML receives stable project names without altering relative, Windows, or UNC paths. 🚀

### 📊 Key Changes
- Added `_clearml_project_name()` in `ultralytics/utils/callbacks/clearml.py` to normalize leading POSIX-style slashes in `trainer.args.project`.
- Updated ClearML task initialization to use the normalized `project_name` instead of passing `trainer.args.project` directly.
- Preserved existing behavior for relative paths, Windows drive paths, and UNC-style paths to keep the fix narrowly scoped. 🛡️
- Added a new `tests/test_clearml.py` suite covering:
- normalization behavior across empty, relative, POSIX, Windows, and UNC path inputs
- correct `Task.init()` argument passing during `on_pretrain_routine_start()`
- graceful handling of `connect()` failures with a warning instead of raising

### 🎯 Purpose & Impact
- Resolves a ClearML inconsistency when Ultralytics output directories use absolute POSIX paths. ✅
- Improves reliability of ClearML project creation and lookup during training runs.
- Reduces regression risk with focused unit tests around callback initialization behavior.
- Keeps the change low-risk by avoiding broader path rewriting outside the reported POSIX case. 🔍

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @MUKILAN0608, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help with review readiness, and an engineer will assist you shortly. Please review the checklist below for a smooth integration 😊

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — DISMISSED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The change looks clean and appropriately scoped. The normalization helper matches the stated intent, and the new tests cover the key POSIX, relative, Windows/UNC, and connect-failure paths. I did not find any actionable issues in this diff.

**Comment by MUKILAN0608** [comment]
I have read the CLA Document and I sign the CLA.

**Comment by MUKILAN0608** [comment]
I have read the CLA Document and I sign the CLA.

**Comment by MUKILAN0608** [comment]
CLA signed.

File: ultralytics/utils/callbacks/clearml.py [hunk 1]
```diff
@@ -14,6 +14,20 @@
...
+def _clearml_project_name(project) -> str:
+    """Convert the Ultralytics project argument into a ClearML project name.
+
+    Ultralytics commonly uses ``project`` as an output directory, while ClearML interprets ``project_name`` as a
+    namespace. Only leading POSIX-style slashes are normalized to avoid the ClearML project lookup/create inconsistency
+    reported in Issue #24820. Windows and UNC-style paths are intentionally left unchanged to keep this fix narrowly
+    scoped.
+    """
+    project_name = str(project or "Ultralytics")
+    if project_name.startswith("/") and (not project_name.startswith("//") or project_name.startswith("///")):
```
Comment (UltralyticsAssistant) [inline ultralytics/utils/callbacks/clearml.py]: 💡 **MEDIUM**: This condition preserves every path that starts with exactly `//`, but only UNC paths should be exempt from normalization. A POSIX absolute path like `//tmp/test` will still be forwarded to ClearML with leading slashes, so the original project lookup/create inconsistency remains for that valid absolute-path form. Consider detecting UNC paths more specifically instead of treating all `//...` values as UNC.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review 2

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Good targeted fix and solid regression coverage overall. One logic gap remains in `_clearml_project_name()`: it currently skips normalization for any `//...` path, which leaves some POSIX absolute paths unhandled.

💬 Posted 1 inline comment

**Comment by MUKILAN0608** [comment]
recheck

**Comment by MUKILAN0608** [comment]
I have read the CLA Document and I sign the CLA

**Comment by MUKILAN0608** [comment]
recheck

**Comment by glenn-jocher** [comment]
Thanks — your CLA acknowledgment is visible now, so there’s nothing else you need to do on that part; the PR can proceed through the normal checks and review flow.

**Comment by glenn-jocher** [comment]
Thanks @MUKILAN0608. I reduced this to the direct one-line ClearML project normalization in the callback and merged it. The larger helper/test scaffolding was not needed for this narrow fix.
